src/passes/Print.cpp

```diff
@@ -2076,7 +2076,10 @@ struct PrintExpressionContents
     o << std::max(curr->tuple->type.size(), size_t(2)) << " ";
     o << curr->index;
   }
-  void visitRefI31(RefI31* curr) { printMedium(o, "ref.i31"); }
+  void visitRefI31(RefI31* curr) {
+    printMedium(
+      o, curr->type.getHeapType().isShared() ? "ref.i31_shared" : "ref.i31");
```

Comment:
I think you may need to guard here against `curr->type` being unreachable (when the i32 child is unreachable), in which case `getHeapType` errors.